js/site-search.js

```diff
@@ -1,41 +1,79 @@
 namespacer('bcpl');
 
-bcpl.siteSearch = (($) => {
+bcpl.siteSearch = (($, window, constants) => {
 	const siteSearchTabSelector = '.search-button';
 	const siteSearchInputSelector = '#site-search-input';
-	const siteSearchClearIconSelector = '.site-search-input-container .fa-times';
 	const siteSearchSearchIconSelector = '.site-search-input-container .fa-search';
+	const searchButtonCatalogSelector = '.search-button-catalog';
+	const searchButtonEventsSelector = '.search-button-events';
+	const searchButtonWebsiteSelector = '.search-button-website';
+	const searchAction = {};
 
 	const onSearchTabClick = (clickEvent) => {
-		const $searchBtn = $(clickEvent.target);
+		const $searchBtn = $(clickEvent.currentTarget);
 		$searchBtn
 			.siblings().removeClass('active').end()
 			.addClass('active');
 	};
 
-	const onSearchClearBtnClick = () => {
-		$(siteSearchInputSelector)
-			.val('')
-			.trigger('keyup')
-			.focus();
+	const onSearchCatalogClick = () => {
+		searchAction.search = () => { searchCatalog(window); };
 	};
 
-	const onSearchInputKeyup = (keyupEvent) => {
-		const $searchInput = $(keyupEvent.target);
-		const doesSearchHaveValue = $searchInput.val();
-		let $elmToHide = $(siteSearchSearchIconSelector);
-		let $elmToShow = $(siteSearchClearIconSelector);
+	const onSearchIconClick = () => {
+		const searchTerms = getSearchTerms();
 
-		if (!doesSearchHaveValue) {
-			$elmToHide = $(siteSearchClearIconSelector);
-			$elmToShow = $(siteSearchSearchIconSelector);
+		if (searchAction && searchAction.search && searchTerms.length) {
+			searchAction.search();
 		}
+	};
+
+	const onSiteSearchKeyup = (keyupEvent) => {
+		const keyCode = keyupEvent.which || keyupEvent.keyCode;
+
+		if (keyCode === bcpl.constants.keyCodes.enter) {
+			const searchTerms = getSearchTerms();
+
+			if (searchAction && searchAction.search && searchTerms.length) {
+				searchAction.search();
+			}
+		}
+	};
+
+	const searchCatalog = (activeWindow) => {
+		const searchTerms = getSearchTerms();
+
+		if (searchTerms.length) {
+			const searchUrl = constants.search.urls.catalog;
+			activeWindow.location.href = `${searchUrl}${searchTerms}`; // eslint-disable-line 			
+		}
+	};
 
-		$elmToHide.hide();
-		$elmToShow.show();
+	const getSearchTerms = () => {
+		const $searchBox = $(siteSearchInputSelector);
+		const searchTerms = $searchBox.val();
+		const trimmedSearchTerms = searchTerms.trim();
+		const encodedSearchTerms = encodeURIComponent(trimmedSearchTerms);
+
+		return encodedSearchTerms;
 	};
```

Comment:
you could write this this way.

```javacript
const getSearchTerms = () => {
      const searchTerms = $(siteSearchInputSelector).val();
      return searchTerms ? encodeURIComponent(searchTerms.trim()) : null; // this could also be : "", not sure what you prefer
};
```